src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java

```diff
@@ -149,7 +180,24 @@ public void testImpl(TestPipeline.TestContext context) throws Exception {
 
                 GtfsRealtime.FeedMessage feedMessage = GtfsRealtime.FeedMessage.parseFrom(received.getData());
                 assertNotNull(feedMessage);
-                //validateCancellationPayload(feedMessage, dvjId, msg.timestamp, route, direction, dateTime);
+                assertEquals(1, feedMessage.getEntityCount());
+                GtfsRealtime.TripUpdate tu = feedMessage.getEntity(0).getTripUpdate();
+                assertNotNull(tu);
+
+                assertEquals(1, tu.getStopTimeUpdateCount());
+                GtfsRealtime.TripUpdate.StopTimeUpdate update = tu.getStopTimeUpdate(0);
+                assertNotNull(update);
+                assertTrue(update.hasArrival());
+
+                // We add departure also which should be the same as arrival?
+                //assertFalse(update.hasDeparture());
```

Comment:
This assert is not needed?

Reply:
actually this should have it's own test completely. Thanx for noticing.

Reply:
Validated from code that yes, we always have both arrival and departure. -> added testcase for both arrival and departure events. 

requires some refactoring but I'll do that in my next pr which is on the way.